src/test/regress/sql/single_node.sql

```diff
@@ -41,6 +41,30 @@ SELECT * FROM test ORDER BY x;
 UPDATE test SET y = y + 1 RETURNING *;
 WITH cte_1 AS (UPDATE test SET y = y - 1 RETURNING *) SELECT * FROM cte_1 ORDER BY 1,2;
 
+-- Test upsert with constraint
+CREATE TABLE upsert_test
+(
+	part_key int UNIQUE,
+	other_col int,
+	third_col int
+);
+-- distribute the table and create shards
+SELECT create_distributed_table('upsert_test', 'part_key');
```

Comment:
might be better if we can add similar tests with reference tables & citus local tables

Reply:
agreed, that'd be good to have a test with citus local tables. Can you work with @onurctirtir   to come up with?

Also, some multi-shard queries would be good (the indentation is off, please fix):
```SQL
BEGIN;
	-- force local execution
	SELECT count(*) FROM upsert_test WHERE part_key = 1;

	-- multi-shard pushdown query that goes through local execution 
	 INSERT INTO upsert_test (part_key, other_col) SELECT part_key, other_col FROM upsert_test ON CONFLICT ON CONSTRAINT upsert_test_part_key_key DO NOTHING RETURNING *;

	-- multi-shard repartition query that goes through local execution 

	 INSERT INTO upsert_test (part_key, other_col) SELECT part_key + 1, other_col FROM upsert_test ON CONFLICT ON CONSTRAINT upsert_test_part_key_key DO NOTHING RETURNING *;

	-- multi-shard pull-to-coordinator query that goes through local execution 

	 INSERT INTO upsert_test (part_key, other_col) SELECT part_key, other_col FROM upsert_test LIMIT 100 ON CONFLICT ON CONSTRAINT upsert_test_part_key_key DO NOTHING RETURNING *;
COMMIT;
```